src/qt/radiobut.cpp

```diff
@@ -76,14 +80,18 @@ bool wxRadioButton::Create( wxWindow *parent,
 
 wxRadioButton::~wxRadioButton()
 {
-    if ( m_qtRadioButton->group() &&
-         m_qtRadioButton->group()->buttons().size() == 1 )
+    if ( m_qtRadioButton->group() && m_qtRadioButton->group()->buttons().size() == 1 )
     {
         // If this button is the only member of the last group, remove the map entry for the group
-        std::map<wxWindow*, QButtonGroup*>::iterator it = m_lastGroup.find( GetParent() );
-        if ( it != m_lastGroup.end() && m_qtRadioButton->group() == it->second )
+        std::map<wxWindow*, QButtonGroup*>::iterator it = GetWindowToButtonGroupMap().find( GetParent() );
+        if ( it != GetWindowToButtonGroupMap().end() && m_qtRadioButton->group() == it->second )
         {
-            m_lastGroup.erase(it);
+            GetWindowToButtonGroupMap().erase(it);
+
+            if( m_qtButtonGroup )
```

Comment:
You can avoid doing this by doing `delete it->second` (or `it->second->deleteLater()`? Before erasing the iterator of course) or `delete m_qtRadioButton->group()`. The former is shorter, so probably preferred.

Reply:
Agreed.